test/UpgradeableProxy.t.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import { Test} from "forge-std/Test.sol";
+
+import { UpgradeableProxy } from "../src/UpgradeableProxy.sol";
+
+import { CounterImplementation } from "./CounterImplementation.sol";
+
+contract ProxyBaseTest is Test {
+
+    CounterImplementation public implementation;
+    UpgradeableProxy      public proxy;
+
+    function setUp() public virtual {
+        implementation = new CounterImplementation();
+
+        proxy = new UpgradeableProxy();
```

Comment:
I usually test for the `Rely()` event in the constructor.

Reply:
Updated and added events assertions for all functions